fontbe/src/gvar.rs

```diff
@@ -0,0 +1,101 @@
+//! Generates a [gvar](https://learn.microsoft.com/en-us/typography/opentype/spec/gvar) table.
+
+use std::cmp::{min, max};
+
+use font_types::{F2Dot14, GlyphId};
+use fontdrasil::orchestration::Work;
+use fontir::coords::NormalizedLocation;
+use write_fonts::{
+    dump_table,
+    tables::{
+        gvar::{GlyphDeltas, GlyphVariations, Gvar},
+        variations::Tuple,
+    },
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, GvarFragment},
+};
+
+struct GvarWork {}
+
+pub fn create_gvar_work() -> Box<BeWork> {
+    Box::new(GvarWork {})
+}
+
+fn to_tuple(loc: &NormalizedLocation) -> Tuple {
+    Tuple::new(
+        loc.iter()
+            .map(|(_, nc)| F2Dot14::from_f32(nc.to_f32()))
+            .collect(),
+    )
+}
+
+fn to_deltas(fragment: &GvarFragment) -> Vec<GlyphDeltas> {
+    fragment
+        .deltas
+        .iter()
+        .filter_map(|(loc, deltas)| {
+            if !loc.has_any_non_zero() {
+                return None;
+            }
+
+            // TODO: add phantoms properly
+            let mut min_x = i16::MAX;
+            let mut max_x = i16::MIN;
+            let mut min_y = i16::MAX;
+            let mut max_y = i16::MIN;
+            
+            
+            // TODO: nice rounding on deltas
+            let mut deltas: Vec<_> = deltas.iter().map(|v| {
+                let (x, y) = (v.x as i16, v.y as i16);
+                min_x = min(min_x, x);
+                max_x = max(max_x, x);
+                min_y = min(min_y, y);
+                max_y = max(max_y, y);                
+
+                (x, y)
+            }).collect();
+
+            // TODO: add phantoms properly
+            deltas.push((min_x, 0));
+            deltas.push((max_x, 0));
+            deltas.push((0, min_y));
+            deltas.push((0, max_y));
+
+            // TODO: what exactly is an intermediate region? Can we ignore it and let Gvar solve for it?
+            Some(GlyphDeltas::new(to_tuple(loc), deltas, None))
```

Comment:
no this is not correct. You can't determine the regions from the locations, the variation model does that, you should be able to get them from it. These are stored in the VariationModel.supports attribute, I think you already have computed those. Deltas are associated with regions, not with locations (the latter convey only the peaks, not the start/end which are only present in the regions' tent definitions), only the absolute master values can be said to have a location.
The python VariationModel has a getDeltasAndSupports method that returns a 2-tuple with the detals (from getDeltas) and the model.supports. You need _both_ to construct gvar tuple variations, as well as any other variation table.

Reply:
I believe the tents are in `influence: Vec<VariationRegion>`.

Reply:
Updated to use influence to get the tents.